pyControl4/room.py

```diff
@@ -0,0 +1,140 @@
+        """
+        Sets the room source, turning on the room if necessary.
+        If audio_only, only the current audio device is changed
+        """
+        await self.director.sendPostRequest(
+            f"/api/v1/items/{self.item_id}/commands",
+            "SELECT_AUDIO_DEVICE" if audio_only else "SELECT_VIDEO_DEVICE",
+            {"deviceid": source_id},
+        )
+
+    async def setAudioSource(self, source_id: int):
+        """Sets the current audio source for the room"""
+        await self._setSource(source_id, audio_only=True)
+
+    async def setVideoAndAudioSource(self, source_id: int):
+        """Sets the current audio and video source for the room"""
+        await self._setSource(source_id, audio_only=False)
+
+    async def getVolume(self) -> int:
+        """Returns the current volume for the room from 0-100"""
+        value = await self.director.getItemVariableValue(self.item_id, "CURRENT_VOLUME")
+        return int(value)
+
+    async def isMuted(self) -> bool:
+        """Returns True if the room is muted"""
+        value = await self.director.getItemVariableValue(self.item_id, "IS_MUTED")
+        return int(value) != 0
+
+    async def setMute(self, muted: bool):
+        """Mute/Unmute the room"""
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "MUTE_ON" if muted else "MUTE_OFF",
+            {},
+        )
+
+    async def toggleMute(self):
+        """Toggle the current mute state for the room"""
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "MUTE_TOGGLE",
+            {},
+        )
+
+    async def setVolume(self, volume: int):
+        """Set the room volume, 0-100"""
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "SET_VOLUME_LEVEL",
+            {"LEVEL": volume},
+        )
+
+    async def setIncrementOrDecrementVolume(self, increase: bool):
+        """Increase/Decrease volume by 1"""
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "PULSE_VOL_UP" if increase else "PULSE_VOL_DOWN",
+            {},
+        )
+
+    async def setPlay(self):
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "PLAY",
+            {},
+        )
+
+    async def setPause(self):
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "PAUSE",
+            {},
+        )
+
+    async def setStop(self):
+        """Stops the currently playing media but does not turn off the room"""
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "STOP",
+            {},
+        )
+
+    async def getAudioDevices(self):
```

Comment:
Could you document/provide an example of what kind of data format is returned by this call?

See here for an example: https://github.com/lawtancool/pyControl4/blob/065485fea8bcac6e4db0d3024596e3fdc1550630/pyControl4/account.py#L157

Reply:
The API call on OS 3.2.3, just returns None for me on both getAudioDevices/getVideoDevices. I do not make use of the functions in the HA integration instead we use the UI. Perhaps we delete these API Calls since they are not used.

Reply:
Deleted.

Reply:
It returns None? Like literal Python `None` or some sort of valid JSON response that just says no devices?

Reply:
Yeah its just None the python literal. That is why I labeled as an invalid API call and deleted but I can put it back with comment that it wasn't valid in OS 3.2.3

Reply:
Done, I committed a change that brings back the functions and adds a warning note.